llvm/lib/CodeGen/MachineSink.cpp

```diff
@@ -100,6 +100,12 @@ static cl::opt<bool>
                                 "register spills"),
                        cl::init(false), cl::Hidden);
 
+static cl::opt<bool> AggressivelySinkInstsIntoCycle(
+    "aggressively-sink-insts-to-avoid-spills",
+    cl::desc("Aggressively sink instructions into cycles to avoid "
+             "register spills"),
+    cl::init(false), cl::Hidden);
```

Comment:
If the intent is to actually use this, should not be a cl::opt. cl::opt is for debugging

Reply:
The alternative being introducing a transparent optimization? I think this is a useful debug mechanism nonetheless.